components/org.wso2.carbon.identity.recovery/src/main/java/org/wso2/carbon/identity/recovery/internal/service/impl/UserAccountRecoveryManager.java

```diff
@@ -284,50 +295,57 @@ public String getUsernameByClaims(Map<String, String> claims, String tenantDomai
                     null);
         }
         int tenantId = IdentityTenantUtil.getTenantId(tenantDomain);
-        String[] resultedUserList = ArrayUtils.EMPTY_STRING_ARRAY;
-        for (String key : claims.keySet()) {
-            String value = claims.get(key);
-            if (StringUtils.isNotEmpty(key) && StringUtils.isNotEmpty(value)) {
-                if (log.isDebugEnabled()) {
-                    String message = String.format("Searching users by claim : %s for value : %s", key, value);
-                    log.debug(message);
+        try {
+            String[] userStoreDomainNames = getDomainNames(tenantId);
+            carbonUM = (AbstractUserStoreManager) getUserStoreManager(tenantId);
+            RealmService realmService = IdentityRecoveryServiceDataHolder.getInstance().getRealmService();
+            ClaimManager claimManager = (ClaimManager) realmService.getTenantUserRealm(IdentityTenantUtil.
+                    getTenantId(tenantDomain)).getClaimManager();
+            ArrayList<org.wso2.carbon.user.core.common.User> resultedUserList = new ArrayList<>();
+
+            for (String domain : userStoreDomainNames) {
+                List<ExpressionCondition> expressionConditionList = new ArrayList<>();
```

Comment:
Can we move this to a seperate method to improve readability

Reply:
Resolved by https://github.com/wso2-extensions/identity-governance/pull/623/commits/7a46067ae7fa2693f21ee7bda490d7c1bd37ef0e